File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.java
class Solution {
    public String interpret(String command) {
        StringBuilder result=new StringBuilder();
        for(int i=0;i<command.length();i++){
            if(command.charAt(i)=='G'){
                result.append("G");
            }else if(i<command.length()-1 && command.charAt(i)=='(' && command.charAt(i+1)==')'){
                result.append("o");
                i++;
            }else{
                result.append("al");
                i+=3;
            }
        }
        return result.toString();
    }
}

File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        res = ""
        i = 0
        while i < len(command):
            if command[i] == "G":
                res += "G"
            elif command[i] == "(" and command[i+1] == ")":
                res += "o"
                i += 1
            else:
                res += "al"
                i += 3
            i += 1
        return res
            